service/frontend.js

```diff
@@ -102,14 +102,15 @@ module.exports = class WebFrontend extends events.EventEmitter {
 
         this._app.use((req, res, next) => {
             let port = ':' + this._app.get('port');
-            if (Config.HAS_REVERSE_PROXY ||
-                (req.protocol === 'http' && port === ':80') ||
-                (req.protocol === 'https' && port === ':443'))
+            if (Config.HAS_REVERSE_PROXY) {
+                let tmp = req.headers['x-forwarded-host'].split(':');
+                if (tmp[1])
+                    port = tmp[1];
```

Comment:
Do you need a colon here?
And also an else clause that sets port = ''

Reply:
yes it's there, the changed lines are from 105 to 112